azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -437,6 +437,12 @@ func resourceArmKubernetesCluster() *schema.Resource {
 				ConflictsWith: []string{"private_link_enabled"},
 			},
 
+			"private_dns_zone_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
```

Comment:
could we add validation that this is a Private DNS Zone ID here too?

```suggestion
				ForceNew: true,
				ValidateFunc: privateDnsValidate.PrivateDnsZoneID,
```